plutus-list/src/PlutusTx/List/Natural.hs

```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE Trustworthy #-}
+
+{- |
+ Module: PlutusTx.List.Natural
+ Copyright: (C) MLabs 2021
+ License: Apache 2.0
+ Maintainer: Koz Ross <koz@mlabs.city>
+ Portability: GHC only
+ Stability: Experimental
+ List functions that works with the 'Natural' type.
+-}
+module PlutusTx.List.Natural (
+  length,
+  replicate,
+  take,
+  drop,
+  splitAt,
+) where
+
+import Data.Monoid (Sum (Sum, getSum))
+import PlutusTx.Natural (Natural, nat)
+import PlutusTx.Prelude hiding (length, take)
+import Prelude qualified ()
+
+{-# INLINEABLE length #-}
```

Comment:
I'm in favour of the 'Haddock, then INLINEABLE, then definition' ordering.  It keeps everything together neatly.

Reply:
@kozross I think probably it's worth mentioning this in the style guide as well so we can keep things consistent